ci/cached-builds/makefile_helper.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+
+import pathlib
+import platform
+import subprocess
+import sys
+
+def exec_makefile(target: str, makefile_dir: pathlib.Path | str, options: list[str] = []) -> str:
```

Comment:
I don't like it. The value of the original function was in the `-nps` args, that made it do something that people who did not read the make docs may not know is possible. This turns the thing into a rather uninteresting wrapper over `subprocess.run`.